core-bundle/src/Twig/Interop/PhpTemplateProxyNodeVisitor.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Twig\Interop;
+
+use Twig\Environment;
+use Twig\Node\BlockNode;
+use Twig\Node\ModuleNode;
+use Twig\Node\Node;
+use Twig\Node\TextNode;
+use Twig\NodeVisitor\AbstractNodeVisitor;
+use Webmozart\PathUtil\Path;
+
+/**
+ * @experimental
+ */
+final class PhpTemplateProxyNodeVisitor extends AbstractNodeVisitor
+{
+    /**
+     * @var string
+     */
+    private $extensionName;
+
+    public function __construct(string $extensionName)
+    {
+        $this->extensionName = $extensionName;
+    }
+
+    public function getPriority(): int
+    {
+        return 0;
+    }
+
+    protected function doEnterNode(Node $node, Environment $env): Node
+    {
+        return $node;
+    }
+
+    protected function doLeaveNode(Node $node, Environment $env): Node
+    {
+        if ($node instanceof ModuleNode && 'html5' === Path::getExtension($node->getTemplateName(), true)) {
+            $this->configurePhpTemplateProxy($node);
+        }
+
+        return $node;
+    }
+
+    /**
+     * We are replacing the module body with a PhpTemplateProxyNode that will
+     * delegate rendering to the Contao framework on the fly. To support blocks
+     * we're also injecting a BlockNode for each block in the original source
+     * that will return the default Contao block placeholder when called.
+     */
+    private function configurePhpTemplateProxy(ModuleNode $node): ModuleNode
```

Comment:
The return value of this method is never used (see line 51). Is this correct?

Reply:
Ah good catch! The method is operating on the reference.

Fixed in f65506b0a9a7186af604ba0051db1bffcd02fe77